### PR TITLE
Offers a new vision of the home page + optimisation relative to course fetching

### PR DESCRIPTION
**Pull request by holoisme**
Hello,
Please see below the new and improved EdWeiss home page, the Calendar now has its own route to breath so it is no longer competing with the rest of the home page for screen space. It also adds a notification button in the upper right corner.

This PR also provides a new way to access the user's courses with the `useCourse(id)` and `useCourses()` hooks, it fetches them  one time at the root of the app tree and passes them via React's context API (a bit like the user context already).

This PR also fixes the rather annoying "new NativeEventEmitter()" warning.

image

### CODE REVIEW

File: model/src/school/courses.ts [hunk 1]
```diff
@@ -30,9 +21,6 @@ export interface Course {
 	periods: CourseTimePeriod[];
 	section: Section;
 	credits: Credits;
-	newAssignments: boolean;
```
Comment (Gwen678) [inline model/src/school/courses.ts:33]: Same as before : How do you plan to handle assignments, including when new ones are added and when the course has started so that students can access them, if you decide to remove these fields: newAssignments, assignments, and started?

**Review by Gwen678 — COMMENTED** [review]
Here's a general comment on the PR: there's not a lot of new code added, but quite a few changes and moves of existing code. Overall, these changes seem well organized and simplify the structure, which is positive. Be careful not to remove code that the team is currently using, though. There's also a lot of commented code, which isn't ideal, as it can create confusion or unintentional omissions. That said, it's a good job overall! The code remains clear and consistent with the project. Lightening up on unnecessary comments would improve readability and maintainability. Good job!

File: edweiss-app/app/(app)/(tabs)/community.tsx [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
-import { useState } from 'react';
-import TText from '@/components/core/TText';
-import TView from '@/components/core/containers/TView';
-
-
-const ExploreTab: ApplicationRoute = () => {
-	const [pageCount, setPageCount] = useState<number>(1);
 
 
+const CommunityTab: ApplicationRoute = () => {
```
Comment (FerdinandGomerChiu) [inline edweiss-app/app/(app)/(tabs)/community.tsx:9]: what is the purpose of the community tab? Is it strictly a sandbox for the team?

Comment (holoisme) [reply edweiss-app/app/(app)/(tabs)/community.tsx:9]: Hello,
For now yes, but in the future it'll be about the campus life (associations, announcements, events...).

File: edweiss-app/model/school/courses.ts [hunk 3]
```diff
@@ -8,15 +8,6 @@ export type CourseID = string & {};
 export type Credits = number & {};
 
 export type Section = "IN" | "COM" | "PH" | "MT";
```
Comment (FerdinandGomerChiu) [inline edweiss-app/model/school/courses.ts:10]: Why did you remove the course colors?

**Review by FerdinandGomerChiu — APPROVED** [review]
Those seem to be sizable changes. Good job.

File: edweiss-app/model/school/courses.ts (hunk 3 above)
Comment (holoisme) [reply edweiss-app/model/school/courses.ts:10]: Hello,
The model files should have nothing to do with colors and display, I moved `courseColors` to the, I believe, more appropriate `constants/Colors.ts` file of the app.

File: edweiss-app/constants/Colors.ts [hunk 4]
```diff
@@ -18,16 +26,16 @@ const Colors = {
 		maroon: "#e64553",
 		peach: "#fe640b",
 		yellow: "#df8e1d",
-		yellowlogo: "#FECB00",
-		clearGreen: "#7FC245",
+		// yellowlogo: "#FECB00",
+		// clearGreen: "#7FC245",
```
[the remark this answers is not in the document]
Comment (holoisme) [reply edweiss-app/constants/Colors.ts:30]: Hello,
The development palette we are using is one of the most popular and implemented palettes in the world (search "Catppuccin ports" you will be amazed).
So if one element doesn't seem to have its place due to its color, maybe renew its design or change the layout? Adding more colors might be a little confusing.

File: edweiss-app/model/school/courses.ts [hunk 5]
```diff
@@ -30,9 +21,6 @@ export interface Course {
 	periods: CourseTimePeriod[];
 	section: Section;
 	credits: Credits;
-	newAssignments: boolean;
```
[the remark this answers is not in the document]
Comment (holoisme) [reply edweiss-app/model/school/courses.ts:33]: Assignments should be handled as a sub-collection as :
1. They can grow indefinitely (the quiz structure is quite length for example)
2. They are not contingent to the Course object (I can use the object without needing the assignments)
It of course is a judgment call, but it will greatly simplify the life of anyone working with assignments.
